framework/src/utils/SlepcSupport.C

```diff
@@ -96,10 +96,53 @@ getSlepcEigenProblemValidParams()
   params.addParam<unsigned int>("n_eigen_pairs", 1, "The number of eigen pairs");
   params.addParam<unsigned int>("n_basis_vectors", 3, "The dimension of eigen subspaces");
 
+  params.addParam<Real>("e_tol", 1.0e-4, "Relative Tolerance for Eigen Solver");
+  params.addParam<unsigned int>("e_max_its", 10000, "Max Linear Iterations for Eigen Solver");
+
   return params;
 }
 
 void
+setSlepcEigenSolverTolerances(EigenProblem & eigen_problem, const InputParameters & params)
+{
+  Moose::PetscSupport::setSinglePetscOption("-eps_tol", stringify(params.get<Real>("e_tol")));
+
+  Moose::PetscSupport::setSinglePetscOption("-eps_max_it",
+                                            stringify(params.get<unsigned int>("e_max_its")));
+
+  // if it is a nonlinear eigenvalue solver, we need to set tolerances for nonlinear solver and
+  // linear solver
+  if (eigen_problem.isNonlinearEigenvalueSolver())
+  {
+    // nonlinear solver tolerances
+    Moose::PetscSupport::setSinglePetscOption("-eps_power_snes_max_it",
+                                              stringify(params.get<unsigned int>("nl_max_its")));
+
+    Moose::PetscSupport::setSinglePetscOption("-eps_power_snes_max_funcs",
+                                              stringify(params.get<unsigned int>("nl_max_funcs")));
+
+    Moose::PetscSupport::setSinglePetscOption("-eps_power_snes_atol",
+                                              stringify(params.get<Real>("nl_abs_tol")));
+
+    Moose::PetscSupport::setSinglePetscOption("-eps_power_snes_rtol",
+                                              stringify(params.get<Real>("nl_rel_tol")));
+
+    Moose::PetscSupport::setSinglePetscOption("-eps_power_snes_stol",
+                                              stringify(params.get<Real>("nl_rel_step_tol")));
+
+    // linear solver
+    Moose::PetscSupport::setSinglePetscOption("-eps_power_ksp_max_it",
+                                              stringify(params.get<unsigned int>("l_max_its")));
+
+    Moose::PetscSupport::setSinglePetscOption("-eps_power_ksp_rtol",
+                                              stringify(params.get<Real>("l_tol")));
+
+    Moose::PetscSupport::setSinglePetscOption("-eps_power_ksp_atol",
+                                              stringify(params.get<Real>("l_abs_step_tol")));
```

Comment:
I suggest to keep these linear iteration control parameters for Newton and create a new set of parameters for controlling power iteration if really necessary. We can do some off-line discussions on this.